front/views/AuthenticationView.tsx

```diff
@@ -32,9 +32,10 @@ const handleSignup = async (username: string, password: string, email: string, a
 	}
 };
 
-const AuthenticationView = () => {
+const AuthenticationView = ({ route, navigation }: any) => {
 	const dispatch = useDispatch();
-	const [mode, setMode] = React.useState("signin" as "signin" | "signup");
+	const isSignup = route.params?.isSignup ?? false;
+	const [mode, setMode] = React.useState((isSignup ? "signup" : "signin") as "signin" | "signup");
```

Comment:
We should avoid the 'as' as much as possible. Even though, here, it insure type safety, please do `useState<'signin' | 'signup'>(...)` instead

Reply:
Ok I can fix this in this pr